src/components/qcResultsUpload/QcResultsUploadForm.vue

```diff
@@ -51,83 +51,87 @@
 import { createQcResultsUploadResource } from '@/services/traction/QcResultsUpload'
 import UploadIcon from '@/icons/UploadIcon.vue'
+import useRootStore from '@/stores'
+import useAlert from '@/composables/useAlert.js'
 
-export default {
-  name: 'QcResultsUploadForm',
-  components: {
-    UploadIcon,
-  },
-  data() {
-    return {
-      usedByOptions: [
-        { value: null, text: 'Please select a option' },
-        { value: 'extraction', text: 'Extraction' },
-      ],
-      file: null,
-      usedBySelected: 'extraction',
-      busy: null,
-      disableUpload: null,
-      uploadSuccessful: null, // possible values: true, false and null
-    }
-  },
-  computed: {
-    api() {
-      return this.$store.getters.api
-    },
-    qcResultUploadsRequest: ({ api }) => api.traction.qc_results_uploads.create,
-    border() {
-      // If upload is successful, highlight the input box in green
-      // If upload is not successful, highlight the input box in red
-      // Otherwise, uploadSuccessful is null and provide no colouring
-      if (this.uploadSuccessful === true) {
-        return 'rounded border border-success'
-      } else if (this.uploadSuccessful === false) {
-        return 'rounded border border-failure'
-      }
-      return 'border-0'
-    },
-  },
-  methods: {
-    reEnable() {
-      this.$refs.qcResultsUploadFile.value = ''
-      this.uploadSuccessful = null
-      this.file = null
-      this.disableUpload = !this.disableUpload
-    },
-    async fileSelected(evt) {
-      if (evt?.target?.files?.length) {
-        this.file = evt.target.files[0]
-      } else {
-        this.file = null
-        return
-      }
-    },
-    async onSubmit() {
-      await this.postCSV()
-    },
-    async postCSV() {
-      // We want to keep the button disabled after every upload, unless refreshed or "Re-enable" clicked
-      this.busy = true
-      this.disableUpload = true
+// Options for the select input
+const usedByOptions = [
+  { value: null, text: 'Please select a option' },
+  { value: 'extraction', text: 'Extraction' },
+]
 
-      const csv = await this.file.text()
-      const data = { csv: csv, usedBySelected: this.usedBySelected }
-      const { success, errors } = await createQcResultsUploadResource(
-        this.qcResultUploadsRequest,
-        data,
-      )
-      this.uploadSuccessful = success
+// State
+const file = ref(null)
+const usedBySelected = ref('extraction')
+const busy = ref(false)
+const disableUpload = ref(false)
+const uploadSuccessful = ref(null) // true, false, or null
 
-      if (success) {
-        this.showAlert(`Successfully imported: ${this.file.name}`, 'success')
-      } else {
-        this.showAlert(errors || 'Unable to upload QC File', 'danger')
-      }
-      this.busy = false
-    },
-  },
+// Refs for DOM elements
+const qcResultsUploadFile = ref(null)
+
+// Store and API
+const rootStore = useRootStore()
+const api = computed(() => rootStore.api)
```

Comment:
does api need to be computed? rootStore.api shouldnt change value.

Reply:
I have tidied it up. No need for most of the code as the request is only used in 1 place.